core/src/main/scala/org/apache/spark/sql/delta/DeltaAnalysis.scala

```diff
@@ -433,7 +460,15 @@ class DeltaAnalysis(session: SparkSession)
           case Some(existingCatalog) => existingCatalog.identifier
           case None => TableIdentifier(path.toString, Some("delta"))
         }
-        val cloneSourceTable = sourceTbl
+        // Reuse the existing schema so that the physical name of columns are consistent between
+        // the existing files and the newly added files.
+        val cloneSourceTable = sourceTbl match {
+          case source: CloneIcebergSource =>
+            // Reuse the existing schema so that the physical name of columns are consistent between
```

Comment:
Yeah, but if you are REPLACE on an existing Delta table and since Iceberg use column mapping, we have to make sure the column mapping metadata match during conversion.